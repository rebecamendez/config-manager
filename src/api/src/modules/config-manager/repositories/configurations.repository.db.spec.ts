import { DataSource } from 'typeorm';
import { GenericContainer, StartedTestContainer } from 'testcontainers';
import { ConfigurationsRepositoryDb } from './configurations.repository.db';

describe('A Configurations db repository', () => {
  let dataSource: DataSource;
  let container: StartedTestContainer;
  let repository: ConfigurationsRepositoryDb;

  beforeAll(async () => {
    const databaseName = 'configuration-repository-test';
    const entities = [__dirname + './../../shared/database/entities/**/*.entity.ts'];

    container = await new GenericContainer('postgres')
      .withEnvironment({ POSTGRES_USER: 'the-user', POSTGRES_PASSWORD: 'the-password', POSTGRES_DB: databaseName })
      .withExposedPorts(5432)
      .withCopyFilesToContainer([{ source: 'init.sql', target: '/docker-entrypoint-initdb.d/init.sql' }])
      .withName(databaseName)
      .start();

    dataSource = new DataSource({
      type: 'postgres',
      host: container.getHost(),
      port: container.getMappedPort(5432),
      username: 'the-user',
      password: 'the-password',
      database: databaseName,
      entities,
      synchronize: false,
      logging: false
    });

    await dataSource.initialize();
    repository = new ConfigurationsRepositoryDb(dataSource);
  });

  afterAll(async () => {
    if (dataSource) {
      await dataSource.destroy();
    }
    if (container) {
      await container.stop();
    }
  });

  it('should get the configurations using the repository', async () => {
    const result = await repository.getConfigurations();
    expect(result).toMatchSnapshot();
  });
});
