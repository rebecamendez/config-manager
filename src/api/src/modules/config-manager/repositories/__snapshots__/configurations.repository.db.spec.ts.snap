// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Configurations db repository should create a configuration using the repository 1`] = `
Configuration {
  "key": "foo",
  "value": "bar",
}
`;

exports[`A Configurations db repository should get a configuration using the repository 1`] = `
Configuration {
  "key": "MyConfig",
  "value": "myValue",
}
`;

exports[`A Configurations db repository should get the configurations using the repository 1`] = `
[
  Configuration {
    "key": "MyConfig",
    "value": "myValue",
  },
]
`;
